File: apps/android/app/src/main/kotlin/io/palyra/mobile/companion/CompanionStore.kt
package io.palyra.mobile.companion

data class MobileVoiceNoteDraft(
    val draftId: String,
    val targetSessionId: String? = null,
    val transcriptText: String,
    val transcriptReviewed: Boolean,
    val durationMs: Long? = null,
)

data class MobileCachedState(
    val approvalsCacheKey: String = "mobile.approvals.cache",
    val sessionsCacheKey: String = "mobile.sessions.cache",
    val inboxCacheKey: String = "mobile.inbox.cache",
    val outboxQueueKey: String = "mobile.voice-note.outbox",
    val revokeMarkerKey: String = "mobile.auth.revoked",
    val revoked: Boolean = false,
    val sessionExpired: Boolean = false,
    val quietHoursEnabled: Boolean = true,
    val voiceNoteOutbox: List<MobileVoiceNoteDraft> = emptyList(),
)

interface CompanionStore {
    fun load(): MobileCachedState
    fun persist(state: MobileCachedState)
    fun markRevoked()
    fun clearRevokeMarker()
}

File: apps/ios/Sources/MobileCompanion/CompanionStore.swift
import Foundation

struct MobileVoiceNoteDraft: Codable, Equatable {
    var draftId: String
    var targetSessionId: String?
    var transcriptText: String
    var transcriptReviewed: Bool
    var durationMs: Int?
}

struct MobileCachedState: Codable, Equatable {
    var approvalsCacheKey: String = "mobile.approvals.cache"
    var sessionsCacheKey: String = "mobile.sessions.cache"
    var inboxCacheKey: String = "mobile.inbox.cache"
    var outboxQueueKey: String = "mobile.voice-note.outbox"
    var revokeMarkerKey: String = "mobile.auth.revoked"
    var revoked: Bool = false
    var sessionExpired: Bool = false
    var quietHoursEnabled: Bool = true
    var voiceNoteOutbox: [MobileVoiceNoteDraft] = []
}

protocol CompanionStore {
    func load() -> MobileCachedState
    func persist(_ state: MobileCachedState)
    func markRevoked()
    func clearRevokeMarker()
}
